src/config/backends/splash.ts

```diff
@@ -0,0 +1,187 @@
+import * as Bluebird from 'bluebird';
+import * as _ from 'lodash';
+import { fs } from 'mz';
+import * as path from 'path';
+
+import * as constants from '../../lib/constants';
+import log from '../../lib/supervisor-console';
+import * as logger from '../../logger';
+import {
+	bootMountPoint,
+	ConfigBackend,
+	ConfigOptions,
+	remountAndWriteAtomic,
+} from './backend';
+
+export class SplashImage extends ConfigBackend {
+	private static readonly BASEPATH = path.join(bootMountPoint, 'splash');
+	private static readonly DEFAULT = path.join(
+		SplashImage.BASEPATH,
+		'balena-logo-default.png',
+	);
+	private static readonly PATHS = [
+		'balena-logo.png',
+		'resin-logo.png',
+	].map((logo) => path.join(SplashImage.BASEPATH, logo));
+	private static readonly PREFIX = `${constants.hostConfigVarPrefix}SPLASH_`;
+	private static readonly CONFIGS = ['image'];
+
+	// Check the header of the image for a png
+	// https://github.com/sindresorhus/is-png/blob/master/index.js
```

Comment:
Adding this citation is nice. We must not take pieces from modules even if it's small without mentioning it.

Reply:
Yeah, I probably will end up adding the dependency